src/main/java/utilities/BaseClass.java

```diff
@@ -4,42 +4,66 @@
 import org.openqa.selenium.JavascriptExecutor;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Action;
+import org.openqa.selenium.interactions.Actions;
 import org.openqa.selenium.support.ui.ExpectedConditions;
 import org.openqa.selenium.support.ui.WebDriverWait;
 
 public class BaseClass {
 	
 	WebDriver driver;
-	
+	Actions action;
+	/*
+	 * Initialize constructor
+	 * */
 	public BaseClass(WebDriver driver) {
 		this.driver=driver;
+		action=new Actions(driver);
+	}
+	
+	/*
+	 * Initialize constructor
+	 * */
+	public void waitCondition(WaitExpectedConditions conditions,WebElement element,int timeOut) {
+		WebDriverWait wait=new WebDriverWait(driver,timeOut);
+		
+		switch (conditions)
+		{
+		case PRESENCE_OF_ELEMENT:
+							wait.until(ExpectedConditions.visibilityOf(element));
+							break;
+		case ELEMENT_TO_BE_CLICKABLE: 
+						  wait.until(ExpectedConditions.elementToBeClickable(element));
+						  break;	
+		}
 	}
+	
 	public void clickBtn(WebElement button) {
+
+		waitCondition(WaitExpectedConditions.ELEMENT_TO_BE_CLICKABLE, button, 30);
 		 button.click();
 	}
 	public void enterInput(WebElement inputField,String inputValue) {
+		waitCondition(WaitExpectedConditions.PRESENCE_OF_ELEMENT, inputField, 30);
 		inputField.sendKeys(inputValue);
 	}
 	public void clearText(WebElement inputField) {
+		waitCondition(WaitExpectedConditions.PRESENCE_OF_ELEMENT, inputField, 30);
 		inputField.clear();
 	}
 	
 	public void Scroll() {
```

Comment:
create function scrollToElement which will scroll to particular webElement instead of providing random no.